pkg/kubelet/kubelet.go

```diff
@@ -2029,6 +2029,15 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 	result := kl.containerRuntime.SyncPod(sctx, pod, podStatus, pullSecrets, kl.backOff)
 	kl.reasonCache.Update(pod.UID, result)
 	if err := result.Error(); err != nil {
+		// If we have any image pull errors, report any failed pull secrets
+		if len(failedPullSecretNames) > 0 {
+			for _, r := range result.SyncResults {
+				if errors.Is(r.Error, images.ErrImagePull) {
```

Comment:
Do we need to send an event for all failures or can we merge them?

Reply:
Good catch, if we have two failed pulls from two different containers, this would emit the same warning event twice. I'll break the loop on the first one found.